flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -82,6 +88,7 @@ public PartitionOperator(DataSet<T> input, Keys<T> pKeys, Partitioner<?> customP
 		this.pKeys = pKeys;
 		this.partitionLocationName = partitionLocationName;
 		this.customPartitioner = customPartitioner;
+		this.distribution = distribution;
```

Comment:
Other `pMethod` like `hashPartition` and `customPartition` don't need the `distribution` and the `distribution` use its default value: `null`, i think no check here also can make sense.